### ISSUE TITLE
Improve Markdown Support - blockquotes with special markers parse wrongly

### ISSUE THREAD

**Issue by igormironchik** [doxygen/doxygen#10902]
**Describe the bug**

Blockquote with special mark for highlighting (like `[!NOTE]`) parses wrongly.

> [!WARNING]
>
> Have a look at emphasis in second paragraph too. You have there issue too.

**Screenshots**

![quote](https://github.com/doxygen/doxygen/assets/11455533/1f89e97c-b69e-4a81-b7bf-179ba0c4fee5)

**To Reproduce**

[Doxygen.zip](https://github.com/doxygen/doxygen/files/15469937/Doxygen.zip)

**Expected behavior**

I'd like to see all other stuff from highlighted blockquote parsed and rendered correctly in it.

**Version**

```
1.12.0 (3d494140a674a8344c165a559d841166bbf3b80f*)
```

```
No LSB modules are available.
Distributor ID: Neon
Description:    KDE neon 6.0
Release:        22.04
Codename:       jammy
```

I've built `Doxygen` from sources.

**Comment by albert-github** [comment]
So you would like to have something like:

![image](https://github.com/doxygen/doxygen/assets/5223533/1db4f5b4-6eaa-4ba1-b049-d3476d3442e2)

(the fact that ``*`markdown-it (Rust)` measurement done with `markdown_it::plugins::extra`*`` is not as emphasis text is already mentioned in another issue).

The problem is that the construction is replaced by a `\note` command and this ends with an empty line ( in this case this is the line with just `>` in the example).

I see 2 workarounds:
```
> [!NOTE]
> *This measurement done with test file in [markdown-it (Rust)](https://github.com/markdown-it-rust/markdown-it)*

> [!NOTE]
> *`markdown-it (Rust)` measurement done with `markdown_it::plugins::extra`*

> [!NOTE]
> **Keep in mind that `markdown-it (Rust)` doesn't support all GFM extensions, so it's not honest
> to compare. `md4qt` supports GitHub's autolinks extension, that `markdown-it (Rust)` doesn't, and
> this requires ~500 microseconds, for example...**
```
note: the extra real empty line and the replacement of the `>` line with a new `> [!NOTE]`

and 
```
> [!NOTE]
> *This measurement done with test file in [markdown-it (Rust)](https://github.com/markdown-it-rust/markdown-it)*<br>
> *`markdown-it (Rust)` measurement done with `markdown_it::plugins::extra`*<br>
> **Keep in mind that `markdown-it (Rust)` doesn't support all GFM extensions, so it's not honest
> to compare. `md4qt` supports GitHub's autolinks extension, that `markdown-it (Rust)` doesn't, and
> this requires ~500 microseconds, for example...**
```
note: the missing "empty lines" and the addition of `<br>`

Though I still think this is something doxygen should fix.

**Comment by albert-github** [comment]
I've just pushed a proposed patch, pull request #10903

**Comment by albert-github** [comment]
Code has been integrated in master on GitHub (please don't close the issue as this will be done at the moment of an official release).

**Comment by igormironchik** [comment]
I confirm that this issue is solved. But after `Note` mark you have extra `<br>` that is not needed. Look at the screenshot. First time there should be one `<br>` in `HTML`.

![Screenshot_20240603_121807](https://github.com/doxygen/doxygen/assets/11455533/d5d7b99c-41a7-4584-84b3-5a7b6e8726a2)

It will be better with one `<br>` after `Note`...

![Screenshot_20240603_122040](https://github.com/doxygen/doxygen/assets/11455533/81e2f1a2-f1c5-4f57-b84d-397589eda2dd)

**Comment by igormironchik** [comment]
Think more complex. 1 `<br>` should be only after first string with `Note`, but all other separators should be with 2 `<br>`. In this case highlighted blockquotes will be delicious.

**Comment by albert-github** [comment]
In your comment you already wrote:
> But after Note mark you have extra <br> that is not needed.

I unfortunately missed that and indeed I think the output improves when having the `2 <br>` layout except for the part after the `Note`.

**Comment by igormironchik** [comment]
You will get:

![1](https://github.com/doxygen/doxygen/assets/11455533/a53c5907-3018-4ee5-9a77-66fd2aff4d0b)

Mmmmmmm.....

**Comment by albert-github** [comment]
I now had a look in the input file:
```
> [!NOTE]
>
> *This measurement done with test file in [markdown-it (Rust)](https://github.com/markdown-it-rust/markdown-it)*
>
> *`markdown-it (Rust)` measurement done with `markdown_it::plugins::extra`*
>
> **Keep in mind that `markdown-it (Rust)` doesn't support all GFM extensions, so it's not honest
> to compare. `md4qt` supports GitHub's autolinks extension, that `markdown-it (Rust)` doesn't, and
> this requires ~500 microseconds, for example...**
```

and here we see an explicit empty line after the `> [!NOTE]`, so I think the `2 <br>` is justified here. When we would remove the "empty line" after the `> [!NOTE]` you will get:
![image](https://github.com/doxygen/doxygen/assets/5223533/3bb4888f-3927-4c28-b2a1-3c0c2372b04d)

which in my opinion is even nicer.

Still I think `1 <br>` after the note would also be nice.

**Comment by igormironchik** [comment]
Anyway you replace `[!NOTE]` in first paragraph with `<dl class="section note"><dt>Note</dt>`. I guess that some `CSS` or something else should add margin after it. And in `<dd>` first empty lines should be just ignored.

**Comment by doxygen** [comment]
I still see some spacing differences between the way GitHub and doxygen render the notes.

Example
```md
> [!NOTE]
>
> Text
>
> > Quotation inside
>
> Too much space above this line!
>
> Event more text

Some text

> [!NOTE]
>
>
>
> Too much space above this line
>
> And one more line

> [!NOTE]
>
> Not a new note!
```


GitHub:

> [!NOTE]
>
> Text
>
> > Quotation inside
>
> Too much space above this line!
>
> Event more text

Some text

> [!NOTE]
>
>
>
> Too much space above this line
>
> And one more line

> [!NOTE]
>
> Not a new note!

Doxygen:
<img width="1216" alt="image" src="https://github.com/doxygen/doxygen/assets/1300762/5a9863dc-34f3-4695-8b87-a525d197e957">

**Comment by albert-github** [comment]
1. I've been thinking about this as well, but I left it in on purpose as the user explicitly specified multiple empty lines. Maybe we should make it dependent on `MARKDOWN_ID_STYLE` when this is `DOXYGEN` we use the explicit extra space and in case of `GITHUB` we reduce the white space.
2. Second part is about the splitting of the notes. Doxygen has explicitly stated that consecutive notes are taken together, maybe we should make a split here as well (will be more difficult though).

@doxygen what is your preference?

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.12.0.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
